app/views/devise/registrations/new.html.erb

```diff
@@ -1,31 +1,72 @@
 <h2>Sign Up</h2>
 
-<%= form_for(resource, :as => resource_name, :url => registration_path(resource_name)) do |f| %>
-  <%= render "devise/shared/error_messages", resource: resource %>
+<% content_for :stylesheets do %>
+  <%= stylesheet_link_tag "signup_form" %>
+<% end %>
 
+<% content_for :javascripts do %>
+  <%= javascript_include_tag "signup_form" %>
+<% end %>
+
+<%= form_for(resource, :as => resource_name, :url => registration_path(resource_name)) do |f| %>
 	<div class="input-field">
-	  <%= f.text_field :first_name, :autofocus => true %>
+		<%= f.text_field :first_name, :title => "", :pattern => ".+", :required => true, :autofocus => true %>
```

Comment:
If you are setting `:title => ""` to suppress the hover message, then you should be consistent with it for the other fields

Reply:
Also, what is the purpose of `:pattern => ".+"`? If `required` is `true`, I think that would be implied, and if `required` is `false`, then it serves no purpose?

Likewise for `last_name`